spec/features/reset_password_spec.rb

```diff
@@ -2,10 +2,30 @@
 
 require_relative './add_reset_password_shared_examples'
 
-feature 'User resets password', js: true do
+feature 'Password reset', js: true do
 
   it_behaves_like "add_reset_password_shared_examples", :reset
 
+  scenario 'while still logged in – user is not stuck in a loop' do
+    # shouldn't ask to reauthenticate when they forgot their password and are trying to reset it
+    # issue: https://github.com/openstax/business-intel/issues/550
+    user = create_user 'user'
+    create_email_address_for user, 'user@example.com'
+    log_in('user','password')
+
+    Timecop.freeze(Time.now + RequireRecentSignin::REAUTHENTICATE_AFTER) do
+      find('[data-provider=identity] .edit').click
+      expect(page).to have_content(t :"sessions.reauthenticate.page_heading")
+      click_link(t :"sessions.authenticate_options.reset_password")
+      expect(page).to have_content(t(:'identities.send_reset.we_sent_email', emails: 'user@example.com'))
+
+      open_email('user@example.com')
+      password_reset_link = get_path_from_absolute_link(current_email, 'a')
+      visit password_reset_link
+      expect(page).not_to(have_current_path(reauthenticate_path))
```

Comment:
Would be nice to also assert that you are in the path/url that renders the "reset password" form.

Reply:
Added ✅